### PR TITLE
Add result plot

### PR DESCRIPTION
**Pull request by martinRenou**
Adds plot to the analysis part of the application (central pane). This plot shares the analysis model with the result table. Every time new data are pushed in the model, the plot is updated automatically. You can choose which data to display for the axises of the plot.

### CODE REVIEW

File: force_wfmanager/central_pane/analysis.py [hunk 1]
```diff
@@ -23,6 +25,9 @@ class Analysis(TraitsTaskPane):
     #: Selected step, used for highlighting in the table/plot
     selected_step_index = Int(None)
 
+    #: The Pareto Front view
+    pareto_front = Instance(ParetoFront)
```
Comment (stefanoborini) [inline force_wfmanager/central_pane/analysis.py]: just call it plot, it's not always guaranteed to display a pareto front.

File: force_wfmanager/central_pane/pareto_front.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    ))
+
+    def _get_plot(self):
+        if self.x is None or self.y is None:
+            return None
+
+        x_index = self.value_names.index(self.x)
+        y_index = self.value_names.index(self.y)
+
+        plot_data = ArrayPlotData()
```
Comment (stefanoborini) [inline force_wfmanager/central_pane/pareto_front.py]: try to see if you can add the points without having to recompute the array every time

File: force_wfmanager/central_pane/pareto_front.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    @on_trait_change('evaluation_steps')
+    def update_data_arrays(self):
+        self.data_arrays = self._compute_data_arrays()
+
+    def _compute_data_arrays(self):
+        if len(self.evaluation_steps) == 0:
+            return []
+
+        data_dim = len(self.evaluation_steps[0])
```
Comment (stefanoborini) [inline force_wfmanager/central_pane/pareto_front.py]: just append the difference of the number of values.

File: force_wfmanager/central_pane/pareto_front.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+from traitsui.api import View, UItem, Item, VGroup, HGroup
+
+from enable.api import Component, ComponentEditor
+
+from chaco.api import ArrayPlotData, Plot
+
+
+class ParetoFront(HasStrictTraits):
+    #: List of parameter names
+    value_names = List(Str)
```
Comment (stefanoborini) [inline force_wfmanager/central_pane/pareto_front.py]: put this one and the evaluation steps in the common model and then pass the model as a reference. You are getting copies here.

File: requirements/requirements.txt [hunk 5]
```diff
@@ -1,3 +1,5 @@
-envisage >= 4.6.0
-stevedore >= 1.24.0
+envisage == 4.6.0
+stevedore == 1.24.0
+# Needed by chaco
```
Comment (stefanoborini) [inline requirements/requirements.txt]: remove this comment and add it to the edm requirements.

**Comment by martinRenou** [comment]
The `Analysis` class should be renamed `AnalysisModel`.  In the `CentralPane` class I should have an `AnalysisModel` instance, a `Plot` instance and a `ResultTable` instance, the first being the model and the last two being the views. This also means that the instances of `Plot` and `ResultTable` currently in the `Analysis` class should go away.

File: force_wfmanager/central_pane/plot.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            bgcolor="white")
+
+        plot.title = "Plot"
+        plot.line_width = 1
+        plot.padding = 50
+
+        return plot
+
+    def _get_value_names(self):
+        # TODO: Filter value names per type (we do not support string
```
Comment (stefanoborini) [inline force_wfmanager/central_pane/plot.py]: Open issue.

File: force_wfmanager/central_pane/plot.py [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+        for evaluation_step in new_evaluation_steps:
+            for index in range(self.data_dim):
+                self.data_arrays[index].append(evaluation_step[index])
+
+        # Update plot data
+        self._update_plot_data()
+
+    @on_trait_change('x,y')
+    def _update_plot_data(self):
+        if self.x is None or self.y is None:
```
Comment (stefanoborini) [inline force_wfmanager/central_pane/plot.py]: verify this behavior in a test case.

File: force_wfmanager/central_pane/plot.py [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+            self.plot_visible = False
```
Comment (stefanoborini) [inline force_wfmanager/central_pane/plot.py]: leave the plot visible.

File: force_wfmanager/central_pane/plot.py [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+    data_arrays = List(List())
+
+    #: The plot data
+    plot_data = Instance(ArrayPlotData)
+
+    #: The 2D plot
+    plot = Instance(Component)
+
+    #: Boolean defining if the plot if visible or not
+    plot_visible = Bool(False)
```
Comment (stefanoborini) [inline force_wfmanager/central_pane/plot.py]: you can remove this.

File: force_wfmanager/central_pane/plot.py [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+
+        # If the number of evaluation steps is less than the number of element
+        # in the data arrays, it certainly means that the model has been
+        # reinitialized. The only thing we can do is recompute the data arrays.
+        if len(evaluation_steps) < len(self.data_arrays[0]):
+            for data_array in self.data_arrays:
+                data_array[:] = []
+
+        # Update the data arrays with the newly added evaluation_steps
+        new_evaluation_steps = evaluation_steps[len(self.data_arrays[0]):]
```
Comment (stefanoborini) [inline force_wfmanager/central_pane/plot.py]: data_arrays can be private.

**Pull request merged**